proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+      // in case it belongs in the next token.
+      preserveCurrent();
+      String theText = ":";
+      for_loop : for (;;) {
+        getChar();
+        switch (currChar) {
+          case 'r':
+          case 'l':
+          case 'c':
+          case 't':
+          case 'u':
+          case 'x':
+          case '0':
+          case '1':
+          case '2':
+          case '3':
+          case '4':
+          case '5':
+          case '6':
+          case '7':
+          case '8':
+          case '9':
+            theText += (char) currInt;
+            isStringAttributes = true;
+            break;
+          default:
+            break for_loop;
+        }
+      }
+      // either string attributes, or the preserved colon
+      // goes into the next token.
+      if (isStringAttributes) {
+        append(theText);
+        preserveDrop();
+      }
+    } // currChar==':'
+
+    return makeToken(PreprocessorParser.QSTRING);
+  }
+
+  ProToken digitStart() throws IOException {
+    LOGGER.trace("Entering digitStart()");
+
+    int ttype = PreprocessorParser.NUMBER;
+    for_loop : for (;;) {
+      switch (currChar) {
+        case '0':
+        case '1':
+        case '2':
+        case '3':
+        case '4':
+        case '5':
+        case '6':
+        case '7':
+        case '8':
+        case '9':
+          append();
+          getChar();
+          break;
+        case 'a':
+        case 'b':
+        case 'c':
+        case 'd':
+        case 'e':
+        case 'f':
+        case 'g':
+        case 'h':
+        case 'i':
+        case 'j':
+        case 'k':
+        case 'l':
+        case 'm':
+        case 'n':
+        case 'o':
+        case 'p':
+        case 'q':
+        case 'r':
+        case 's':
+        case 't':
+        case 'u':
+        case 'v':
+        case 'w':
+        case 'x':
+        case 'y':
+        case 'z':
+        case '#':
+        case '$':
+        case '%':
+        case '&':
+        case '_':
+          append();
+          getChar();
+          if (ttype != PreprocessorParser.FILENAME)
+            ttype = PreprocessorParser.ID;
+          break;
+        // We don't know here if the plus or minus is in the middle or at the end.
+        // Don't change ttype.
+        case '+':
+        case '-':
+          append();
+          getChar();
+          break;
+        case '/':
+          append();
+          getChar();
+          if (ttype == PreprocessorParser.NUMBER)
+            ttype = PreprocessorParser.LEXDATE;
+          break;
+        case '\\':
+          append();
+          getChar();
+          ttype = PreprocessorParser.FILENAME;
+          break;
+        case '.':
+          if (prepro.isNameDot()) {
+            append();
+            getChar();
+            break;
+          } else
+            break for_loop;
+        default:
+          break for_loop;
+      }
+    }
+    return makeToken(ttype);
+  }
+
+  ProToken plusMinusStart(int inputType) throws IOException {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "plusMinusStart" is 51 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)